### ISSUE TITLE
Next release

### ISSUE THREAD

**Issue by tkapis** [KozGit/DOOM-3-BFG-VR#186]
Hi Kozgit, this is a fantastic mod but I am waiting patiently to play further because of the problems with saving and quick saves. Have you any idea when the next release will be coming please?

**Comment by KozGit** [comment]
First, Im sorry it's taking long to release the next version, I never thought it would take this long to finish it.  Anyway, there are 3 things Im working on I need to complete  before release. ( Changes to  cutscenes, fix the leaning mechanism, full body orientation. ) The good news is I don't think those will take long to finish.

The bad news - I'm still on vacation until Monday, so can't work on it at all until then.  The really bad news - there might still be  save game issues. 2eyeGuy had been working on those issues, and I actually thought it was resolved.  However, I was testing the lost missions, saved a game, and the game crashed on loading, even with the fixes 2eye had made.

 My current plan is to finish the changes I'm working on, hopefully I can do this by not this coming weekend  but the next. When the changes are done, I'm going to post it here on the issues board so interested people can start testing those  changes before I post it on the release page.  2eyeGuy is MIA, so as soon  as I post it for testing, I'm going to jump into the changes that were made to game saving/loading to start figuring out what's going on. 

Sorry again it's taking so long.

**Comment by tkapis** [comment]
Thanks for your reply.

I will patiently wait for the release because my biggest problem is when I have put weapons in the holsters and I use save or quick save the holsters are empty afterwards.

**Comment by motorsep** [comment]
Are you experiencing save game issues with existing savegames or newly made savegames which were made with your current dev build ?

If former, then I think we can live with that - we'll just re-play the game from start ;)

**Comment by AerowynX** [comment]
i'd hope Doom VFR works better lol game was built from ground up for VR :)

**Comment by motorsep** [comment]
@KozGit  It's barely Wednesday, but how is it going ?

**Comment by KozGit** [comment]
Unfortunately, I've had literally no time to work on this.  I got back from vacation Monday, and it's been nothing but emergencies at the office every day - I haven't had 10 minutes to work on this.  I should have about 2 hours to work on this tomorrow, but then my weekend is booked helping my in-laws with their house.   I just don't seem to have enough hours in the day.

So far, my calendar appears to return to normal next week - I just need to find a few solid hours to wrap this up.  Sorry it's not better news - but I do promise I'll get this done as soon as I can.

**Comment by motorsep** [comment]
No worries, life happens all the time. Thanks for letting community know.

**Comment by jdawgzim** [comment]
KozGit, thanks for the work you put in on this. This mod is awesome and I love the control system you have implemented.

tkapis, I too have the issue where the weapon in the slot disappears but it happens in a lot more situations. When something loads or triggers and when I pick up a power-up or new weapon the slotted weapon disappears.

**Comment by cercata** [comment]
Any news on the roadmap ?

**Comment by jdawgzim** [comment]
This: https://github.com/KozGit/DOOM-3-BFG-VR/graphs/contributors
From the graphs you can tell either people are not committing or efforts have died down in the summer.  So far this is the **best long VR game** I have ever played. Just has lots of minor bugs and some things that could be implemented better.  

I think the best way to help this game is become a **beta tester** and **promoter** for the game in forums. Need more people playing it and properly reporting bugs.  Make sure you can help others perfectly reproduce the bugs and issues you find.

**Comment by AerowynX** [comment]
i got a couple doom 3 bfg keys to giveaway been holding onto them till the next version was put out.. going to do a giveaway on the steamVR facebook group when the next version releases will help get the word out

**Comment by KozGit** [comment]
Wow, thats awesome AerowynX!
  
In regards to the roadmap, I still don't have a date, but I have had a little time the last few days.  I just finished re-working the lean mechanic, so you can access some of the hard to reach consoles without that 'the world is pushing back' feeling.  The screen (optionally) also fades to black now if you put your eyes in a wall/solid.
Now I'm straightening out some changes to how cutscenes are handled, then I need to clean up the code that handles the direction the player model is facing.  
When those are done, I want to soft release it here  - I think there may still be an issue with save games corrupting, and I could use some help from you guys testing it before a general release.

Feel free to post / remind me what you guys think are the most annoying bugs you would like to see addressed - hopefully they've already been taken care of.  ( I have made a fair number of changes locally that haven't been pushed to the repo yet, so there has been more activity than the graphs show.)

Also, sorry again it's taking so long.  I've been crazy busy - just not enough free time.

**Comment by cercata** [comment]
Wow, thans for that roadmap, glad to see you're making progress :)

**Comment by jdawgzim** [comment]
Well since you're asking...

Most annoying issues at top:
* Accessing menu/PDA removes my holstered weapon
* Reloading disables weapon switching (and teleporting when it's on right hand)
* No Vive option for click down to activate pad edges (Moving, weapon switching)
* testSaveGameEnumerate every time I want to load a game (Everytime I die)
* Inconsistency when I can use in-game touch panels. Seems like I have to be looking at them while moving toward them?
* Not being able to teleport in certain areas
* Randomly teleporting through walls/floors/ceilings
* Please add push to activate voice commands button
* Could we get a 3rd weapon holster on our back?  Maybe even grenade holster on the chest? Throw one grenade then switch back to last weapon?

**Comment by KozGit** [comment]
Alright - I fixed the bug where the holstered weapon disappears when opening the pda/system menu.  ( Still need to make sure holstered weapons survive save games/level transitions. )
Added an instant acceleration/deceleration option. Added an optional choke value for the shotgun so it feels a little more useful at normal VR distances.

Still working on the other items - I'll try to keep posting updates here to show progress...

**Comment by CylonSurfer** [comment]
Did you manage to get the player height options working correctly?

**Comment by KozGit** [comment]
Yes, world scale now.works, and there are options for how you want to handle the viewheight,  and and an option to always have the floor in game match the floor in real life.  The only thing I  want to address in the future is changing the height of the player model to reflect actual differ nces in your height. ( This is only an issue when the player body is visible). I cant just scale the whole player model, because it will break all player collision and hit detection.   Currently, the model will crouch some to help but it only goes so far - the camera will start shifting down into the player body if the difference is too great.

**Comment by KozGit** [comment]
Holstered weapons now survive game load/save and level transitions.

**Comment by jdawgzim** [comment]
Awesome! I think I'll wait until the next patch to play Resurrection of Evil. I beat the base game last week. So good!

BUG: While playing Resurrection of Evil the Menu becomes broken. It works for a while then it normally pops up but then disappears. I can still hear it when I navigate it but I can't see it. Also sometimes my flashlight in left hand mode will turn off.  So then I can't access the menu to fix it.  I wonder if it's related to using the Artifact (heart thing) because I've been using that more.

**Comment by jdawgzim** [comment]
FYI: Doom 3 BFG VR # 14 on top 100 Vive games: https://www.vrgamerankings.com/top-100-htc-vive-games  Congrats!

**Comment by tkapis** [comment]
Hi KozGit, any idea when the next patch is being released or maybe just the holster correction could be released please as that is what I am waiting for to play further.

**Comment by jdawgzim** [comment]
Question:
Is there a way to set pulling weapons from the upper/lower back to specific weapons? So setting upper back to always pull rocket launcher and lower back to always pull shotgun. Would be cool too if it'd switch back to weapon you had before if you grab for the weapon you are already holding.

i.e. Hold pistol; Grab for shotgun on lower back; Grab for shotgun on lower back again and get pistol back; Grab for shotgun again; Switch to another weapon (not shotgun); Grab for shotgun again and get shotgun;
